Code/GraphMol/SmilesParse/catch_tests.cpp

```diff
@@ -2896,31 +2896,42 @@ TEST_CASE("Ignore atom map numbers") {
 
 TEST_CASE("Github #7340", "[Reaction][CX][CXSmiles]") {
   SECTION("Test getCXExtensions with a Vector"){
-    const MOL_SPTR_VECT mol_sptr_vect = {
-      "CCO* |$;;;_R1$(0,0,0;1.5,0,0;1.5,1.5,0;0,1.5,0)|"_smiles, 
-      "C1CCCCC1 |$;label2;$|"_smiles, 
-      "CC(=O)O |$;label1;$|"_smiles, 
-      "*-C-* |$star_e;;star_e$,Sg:n:1::ht|"_smiles, 
-    };  
-
-    for (const auto& entry : mol_sptr_vect) {
+    // Create the MOL_SPTR_VECT to hold the molecular pointers
+    const std::vector<ROMol *> mol_vect = {
```

Comment:
We try and make sure that there can't be memory leaks in the tests (@ricrogz has done a bunch of work to make sure we stay clean here), so I think it would be better here to have something like this (not tested) both here and below:
```
  const std::vector<std::unique_ptr<ROMol>> mols = {
   "CCO* |$;;;_R1$(0,0,0;1.5,0,0;1.5,1.5,0;0,1.5,0)|"_smiles,
     etc.
   };
  const std::vector<ROMol *> mol_vect(mols.begin(),mol.end());
```

Reply:
Was getting an error for what you suggested, however, I got it to work with using `auto` and a different initialization setup.